drivers/hmis/app/graphql/types/base_audit_event.rb

```diff
@@ -36,13 +36,47 @@ def self.build(node_class, field_permissions: nil, transform_changes: nil)
     field :id, ID, null: false
     field :record_id, ID, null: false, method: :item_id
     field :record_name, String, null: false
+    field :graphql_type, String, null: false
     field :event, HmisSchema::Enums::AuditEventType, null: false
     field :created_at, GraphQL::Types::ISO8601DateTime, null: false
     field :user, Application::User, null: true
     field :object_changes, Types::JsonObject, null: true, description: 'Format is { field: { fieldName: "GQL field name", displayName: "Human readable name", values: [old, new] } }'
 
+    available_filter_options do
+      arg :audit_event_record_type, [ID]
+      arg :user_id, [ID]
+    end
+
     def record_name
-      object.item_type.demodulize.gsub(/^CustomClient/, '').underscore.humanize.titleize
+      case object.item_type
+      when 'Hmis::Hud::CustomAssessment'
+        'Assessment'
+      when 'Hmis::Hud::Assessment'
+        'CE Assessment'
+      when 'Hmis::Hud::Event'
+        'CE Event'
```

Comment:
Are these names defined on the associated classes?  I feel like these are brittle (and down on line 74.)  Can we ask `Hmis::Hud::Assessment` for some sort of title, so if we change them in the future they'll align?

Reply:
Yeah, I wonder if we could do something like this @ttoomey:
* Add a `display_name` attribute to all the classes. `Hmis::Hud::Base` could have a default that returns the demodulized class name, so we only need to override it where necessary
* Add an enum `EnrollmentRecordTypes`. The enrollment audit filter option could look like `arg :audit_event_record_type, [HmisSchema::Enums::EnrollmentRecordType]`. There would be no need for a custom PickListOption, since the frontend filter component will just use the existing Enum.
  *  The Enum values correspond to graphql type names and the descriptions have the `display_name`s from the classes? That way we can maybe also solve the `graphql_type` thing below, and we just need to maintain a mapping between class name -> graphql type name, which could also be done on the model classes.

That's a rough idea, let me know if you want to chat about options.!

Reply:
Discussed with @ttoomey, I agree we can leave it as-is until we know what pattern we need. It likely makes sense to fetch the record type list from the backend, if we want to incorporate specific custom data elements into the list for filtering.